modules/ROOT/pages/redelivery-policy.adoc

```diff
@@ -0,0 +1,64 @@
+= Redelivery Policy
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: anypoint studio, flow, redelivery, policy
+
+A redelivery policy is a filter that helps you conserve resources by limiting the number of times the Mule runtime engine (Mule) executes messages that result in delivery failures. You can add a redelivery policy to some connectors that contain flow sources, such as the xref:connectors::salesforce/salesforce-connector-reference.adoc[Salesforce], xref:connectors::sap/sap-connector-reference.adoc[SAP], and xref:connectors::vm/vm-reference.adoc[VM] connectors. To see whether a connector can use a redelivery policy, see the reference document for that connector.
+
+When you add a redelivery policy to a flow's source, Mule evaluates the received data before it executes the flow's components. If a message delivery fails a specified number of times, the redelivery policy prevents the flow from processing the received data and raises a REDELIVERY_EXHAUSTED error.
+
+Some connectors have their own logic for handling an error thrown by a redelivery policy. In such cases, check the documentation for the connector to which the source belongs.
+
+== Redelivery Policy Configuration Parameters
+
+[%header,cols="20a,20a,20a,60a"]
+|===
+|Field |Type |Default Value |Description
+|*Max Redelivery Count* |Number |5 |Maximum number of times that a message can be redelivered and processed unsuccessfully before raising a MULE:REDELIVERY_EXHAUSTED error.
+|*Use Secure Hash* |Boolean |True |Indicates whether to use a secure hash algorithm to identify a redelivered message.
+|*Message Digest Algorithm* |String |SHA-256 |Secure hashing algorithm to use for the message. If the payload of the message is a Java object, Mule ignores the *Message Digest Algorithm* value and returns the value that the payload's `hashCode()` returned.
+|*ID Expression* |Expression|- | Defines one or more expressions that determine when a message has been redelivered. This property can be set only if the value of *Use Secure Hash* is False.
+|*Object Store* |String |Object Store created by Mule's default `ObjectStoreManager`. Nonpersistent, with an `entryTtl` of 300 seconds and an `expirationInterval` of 6 seconds. |Object Store in which
+ the redelivery counter for each message is stored. You can configure the Object Store as a reference or an inner element.
+
+For information, see the xref:connectors::object-store/object-store-connector.adoc[Object Store connector documentation].
+|===
+
+== How the Redelivery Policy Works
+
+Each time the source receives a new message, Mule generates a key from the message by using the configured *Message Digest Algorithm* or by evaluating the ID expression.
```

Comment:
`Mule generates a key from the message `-> `Mule identifies the message by generating its key`